api/src/main/java/io/opencensus/metrics/DoubleGaugeMetric.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.metrics;
+
+import io.opencensus.common.ToDoubleFunction;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Double Gauge metric to report instantaneous values. Gauges can go both up and down. For example,
+ * Free memory or Total memory.
+ *
+ * <p>Example 1: Create a Gauge without a labels
+ *
+ * <pre>{@code
+ * class YourClass {
+ *
+ *   private static final MetricRegistry metricRegistry = Metrics.getMetricRegistry();
+ *   DoubleGaugeMetric totalMemory = metricRegistry.addDoubleGaugeMetric(
+ *       "Total_memory", "Total CPU Memory", "1", new ArrayList<LabelKey>());
+ *
+ *   DataPoint defaultDataPoint = totalMemory.getDefaultDataPoint();
+ *
+ *   void doWork() {
+ *      defaultDataPoint.inc();
+ *      // Your code here.
+ *      defaultDataPoint.dec();
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>Example 2: You can also use labels(keys and values) to track different types of metric.
+ *
+ * <pre>{@code
+ * class YourClass {
+ *
+ *   private static final MetricRegistry metricRegistry = Metrics.getMetricRegistry();
+ *
+ *   List<LabelKey> keys = Collections.singletonList(LabelKey.create("Type","desc"));
+ *   DoubleGaugeMetric totalMemory = metricRegistry.addDoubleGaugeMetric(
+ *       "Total_Memory", "Total CPU Memory", "1", keys);
+ *
+ *   List<LabelValue> usedMemory = Collections.singletonList(LabelValue.create("Used"));
+ *   DataPoint dataPoint = totalMemory.addDataPoint(usedMemory);
+ *
+ *   void doSomeWork() {
+ *      dataPoint.set(12.5);
+ *      // Your code here.
+ *      dataPoint.dec();
+ *   }
+ * }
+ * }</pre>
+ *
+ * @since 0.17
+ */
+public abstract class DoubleGaugeMetric {
+
+  /**
+   * Add and returns new Data Point. This is more convenient form when you can want to manually
+   * increase and decrease values as per your service requirements. Must be passed the same number
+   * of label values are were passed to {@link MetricRegistry#addDoubleGaugeMetric} label keys.
+   *
+   * @param labelValues the list of label values.
+   * @return DataPoint the value of single gauge
+   */
+  public abstract DataPoint addDataPoint(List<LabelValue> labelValues);
+
+  /**
+   * Add and returns new Data Point that reports the value of the object after the function. This is
```

Comment:
Adds and returns. Comments should be descriptive ("Opens the file") rather than imperative ("Open the file"), please fix all the comments.

Reply:
Done